Tests/MongoSwiftSyncTests/UnifiedTestRunner/UnifiedOperation.swift

```diff
@@ -0,0 +1,129 @@
+import Foundation
+import MongoSwiftSync
+
+struct UnifiedOperation: Decodable {
+    /// Represents an object on which to perform an operation.
+    enum Object: RawRepresentable, Decodable {
+        /// Used for special test operations.
+        case testRunner
+        /// An entity name e.g. "client0".
+        case entity(String)
+
+        public var rawValue: String {
+            switch self {
+            case .testRunner:
+                return "testRunner"
+            case let .entity(s):
+                return s
+            }
+        }
+
+        public init(rawValue: String) {
+            switch rawValue {
+            case "testRunner":
+                self = .testRunner
+            default:
+                self = .entity(rawValue)
+            }
+        }
+    }
+
+    /// Object on which to perform the operation.
+    let object: Object
+
+    // TODO: SWIFT-913: parse into an operation instance rather than raw name and arguments.
+    let name: String
+    let arguments: BSONDocument?
+
+    /// Expected result of the operation.
+    let result: UnifiedOperationResult?
+
+    private enum CodingKeys: String, CodingKey {
+        case name, object, arguments, expectError, expectResult, saveResultAsEntity
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        self.name = try container.decode(String.self, forKey: .name)
+        self.arguments = try container.decodeIfPresent(BSONDocument.self, forKey: .arguments)
+
+        self.object = try container.decode(Object.self, forKey: .object)
+
+        let singleContainer = try decoder.singleValueContainer()
+        let result = try singleContainer.decode(UnifiedOperationResult.self)
+        guard !result.isEmpty else {
+            self.result = nil
+            return
+        }
+
+        self.result = result
+    }
+}
+
+/// Represents the expected result of an operation.
+enum UnifiedOperationResult: Decodable {
+    /// One or more assertions for an error expected to be raised by the operation.
+    case error(ExpectedError)
+    /// - result: A value corresponding to the expected result of the operation.
+    /// - saveAsEntity: If specified, the actual result returned by the operation (if any) will be saved with this
+    ///       name in the Entity Map.
+    // TODO: SWIFT-913: consider using custom type to represent results
```

Comment:
I need to look into a bit more, but we might want do something similar to `TestOperationResult` from the old runner here, though it would need to be a bit more complex because there can now be the various custom operators e.g. $$unsetOrMatches within the result. if I understand correctly though it looks like the Rust runner just uses its BSON enum here though, so maybe that won't be needed.

Reply:
I think it'll depend how we want to implement matching. I'm okay with just leaving this TODO here and determine the structure here as needed in later PRs.